src/codecoverage/backend/codecoverage_backend/covdir.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+import json
+import os
+
+
+def get_path_coverage(report_path, object_path, max_depth=1):
+    '''
+    Load a covdir report and
+    Recursively format the paths encountered, adding informations relative
+    to file type (file|directory)
+    '''
+    assert os.path.exists(report_path)
+    # TODO: move to ijson to reduce loading time
+    report = json.load(open(report_path))
+
+    # Find the section from the path
+    parts = object_path.split('/')
+    for part in filter(None, parts):
+        if part not in report['children']:
+            raise Exception('Path {} not found in report'.format(object_path))
+        report = report['children'][part]
+
+    def _clean_object(obj, base_path, depth=0):
```

Comment:
I wonder if some of the changes here couldn't be avoided, leaving the onus on the frontend.
E.g. adding the `type` could be unnecessary, as the frontend can check if there are 'children' itself.
Similarly for `path`.

Reply:
I did that to ease the work on the frontend side, but i'm totally ok with leaving the `covdir` format as it is.

Reply:
Yeah let's do that. At least removing `type` and `path`. This way we keep the backend as lean as possible, almost just serving the report file.